syncer/state_syncer.go

```diff
@@ -284,35 +283,24 @@ func (s *Syncer) ensureMeshAgreement(
 			)
 			continue
 		}
-		missing := make(map[types.ATXID]struct{})
-		for _, id := range ed.AtxIDs {
-			if _, ok := missing[id]; ok {
-				continue
-			}
-			hdr, _ := s.cdb.GetAtxHeader(id)
-			if hdr == nil {
-				missing[id] = struct{}{}
-				continue
-			}
-		}
+		missing := s.asCache.GetMissingActiveSet(diffLayer.GetEpoch(), ed.AtxIDs)
```

Comment:
do you know how frequently this can be executed? there will be exclusive lock on this codepath

Reply:
- in steady state, a node resolves hash with the same peer on the same diff hash once every hour (MaxStaleDuration)
- for each sync run (once every minute), it resolves hash diff with N peers, once per peer. N is the number of peers that has different mesh hash.

so the worst case scenario is N times per minute where N capped by number of peers.